application/helpers/frontend_helper.php

```diff
@@ -2082,8 +2082,10 @@ function display_first_page($thissurvey, $aSurveyInfo)
         $thissurvey['EM']['ScriptsAndHiddenInputs'] .= \CHtml::hiddenField('token', $_SESSION['survey_'.$surveyid]['token'], array('id'=>'token'));
     }
 
-    if (isset($loadsecurity)) {
-        $thissurvey['EM']['ScriptsAndHiddenInputs'] .= "\n<input type='hidden' name='loadsecurity' value='$loadsecurity' id='loadsecurity' />\n";
+    /* Get the loadsecurity parameters from captcha only by POST value and set it in current page (1st position) */
+    $loadsecurity = App()->getRequest()->getPost('loadsecurity');
+    if (!empty($loadsecurity)) {
+        $thissurvey['EM']['ScriptsAndHiddenInputs'] .= \CHtml::hiddenField('loadsecurity', $loadsecurity, array('id' => 'loadsecurity'));
```

Comment:
Hm CHtml has built-in protection for XSS issues?

Reply:
You can remove the `\` slash btw, we're already in global namespace. :)

Reply:
Oh same error everywhere. Pah. Pah!

Reply:
> Oh same error everywhere. Pah. Pah!

In 5 and 6

Reply:
> > Hm CHtml has built-in protection for XSS issues?
> 
> yes … encode all value of atributes.

https://www.yiiframework.com/doc/api/1.1/CHtml#renderAttributes-detail